Content.Server/Clothing/ClothingSystem.cs

```diff
@@ -0,0 +1,118 @@
+using System.Threading.Tasks;
+using Content.Server.Clothing.Components;
+using Content.Shared.Actions;
+using Content.Shared.Toggleable;
+using Content.Shared.Inventory;
+using Content.Server.Body.Components;
+using Content.Server.Nutrition.EntitySystems;
+using Content.Server.Atmos.Components;
+using Content.Server.Popups;
+using Robust.Shared.Player;
+using Content.Server.Disease.Components;
+
+namespace Content.Server.Clothing
+{
+    public sealed class ClothingSystem : EntitySystem
+    {
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        [Dependency] private readonly InventorySystem _inventorySystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<ClothingComponent, GetItemActionsEvent>(OnGetActions);
+            SubscribeLocalEvent<ClothingComponent, ToggleActionEvent>(OnActionToggle);
+        }
+
+        private void OnGetActions(EntityUid uid, ClothingComponent component, GetItemActionsEvent args)
+        {
+            if (component.ToggleAction != null)
+            {
+                component.IsToggled = true;
+                args.Actions.Add(component.ToggleAction);
+                //reset on pickup to ensure not equipped in untoggled state
+                ToggleMask(uid, component, default, false);
+            }
+        }
+
+        private void OnActionToggle(EntityUid uid, ClothingComponent component, ToggleActionEvent args)
+        {
+            if (args.Handled)
+                return;
+
+            if ((component.SlotFlags & SlotFlags.MASK) != 0)
```

Comment:
This subscription for mask-specific behavior on one of the most generic components in the game is a bit sussy

Reply:
Yes, I can move it to a separate component. I could try stapling it onto TogglableClothingComponent or just make a new one.